### PR TITLE
Use Typescript class syntax

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: src.ts/agent/config.ts [hunk 1]
```diff
@@ -13,29 +13,25 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-'use strict';
 
 /**
  * @typedef {object} DebugAgentConfig
```
Comment (ofrobots) [inline src.ts/agent/config.ts]: nit: omit this now?

File: src.ts/agent/config.ts [hunk 2]
```diff
@@ -13,29 +13,25 @@
...
  */
-
-module.exports = {
-  // FIXME(ofrobots): presently this is dependent what cwd() is at the time this
-  // file is first required. We should make the default config static.
+export interface DebugAgentConfig {
   /**
    * @property {?string}
```
Comment (ofrobots) [inline src.ts/agent/config.ts]: Here and elsewhere, omit the types from the jsdoc annotation.

File: src.ts/agent/config.ts [hunk 3]
```diff
@@ -13,29 +13,25 @@
...
    * @memberof DebugAgentConfig
```
Comment (ofrobots) [inline src.ts/agent/config.ts]: Here and elsewhere, omit  `@ memberof` annotations.

File: src.ts/agent/config.ts [hunk 4]
```diff
@@ -13,29 +13,25 @@
...
    * @default
```
Comment (ofrobots) [inline src.ts/agent/config.ts]: The `@ default` doesn't make sense anymore.. how do we document the default value?

File: src.ts/agent/debuglet.ts [hunk 5]
```diff
@@ -0,0 +1,774 @@
+/**
```
Comment (ofrobots) [inline src.ts/agent/debuglet.ts]: I don't see the debuglet.js in the patch. Did you miss it? Also, the patch contains renamed files that git doesn't know about being a rename. Is it easy for you to fix this?

File: src.ts/status-message.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  // New status messages cannot be added here.
+  static readonly UNSPECIFIED = 'UNSPECIFIED';
+  static readonly BREAKPOINT_SOURCE_LOCATION =
+                  'BREAKPOINT_SOURCE_LOCATION';
+  static readonly BREAKPOINT_CONDITION = 'BREAKPOINT_CONDITION';
+  static readonly BREAKPOINT_EXPRESSION = 'BREAKPOINT_EXPRESSION';
+  static readonly VARIABLE_NAME = 'VARIABLE_NAME';
+  static readonly VARIABLE_VALUE = 'VARIABLE_VALUE';
+  static readonly BREAKPOINT_AGE = 'BREAKPOINT_AGE';
+}
```
Comment (ofrobots) [inline src.ts/status-message.ts]: nit: missing new line.

File: src.ts/agent/debuglet.ts (hunk 5 above)
Comment (DominicKramer) [inline src.ts/agent/debuglet.ts]: I think I can cherry-pick more smaller changes to fix this.

File: src.ts/debuggee.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  private agentVersion;
+  private labels;
+  private sourceContexts;
+  private statusMessage;
+
+  /**
+   * Creates a Debuggee service object.
+   * @ref https://cloud.google.com/debugger/api/reference/rest/v2/Debuggee
+   *
+   * @param {object} properties - an object with properties to use for Debuggee
```
Comment (ofrobots) [inline src.ts/debuggee.ts]: It would be good to make this an interface. Okay to do this in a follow-on.

**Comment by DominicKramer** [comment]
This PR has been replaced with [PR #285 ](https://github.com/GoogleCloudPlatform/cloud-debug-nodejs/pull/285).

File: src.ts/agent/config.ts (hunk 1 above)
Comment (DominicKramer) [inline src.ts/agent/config.ts]: Fixed in PR #285.

File: src.ts/agent/debuglet.ts (hunk 5 above)
Comment (DominicKramer) [inline src.ts/agent/debuglet.ts]: See PR #285 for the new PR with these changes addressed.

File: src.ts/status-message.ts (hunk 6 above)
Comment (DominicKramer) [inline src.ts/status-message.ts]: Fixed in PR #285.

File: src.ts/debuggee.ts (hunk 7 above)
Comment (DominicKramer) [inline src.ts/debuggee.ts]: An interface for this parameter will be added in a follow-up PR.

**Pull request closed without merging**